lib/valid_email2/dynamic_option_values.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal:true
+
+module ValidEmail2
+  class DynamicOptionValues
+    class << self
+      def blacklist
+        @blacklist ||= Set.new
+      end
+
+      def blacklist=(set)
+        return unless set.is_a?(Set)
+
+        @blacklist = set
+      end
+
+      def blacklist_active_record_query
+        @blacklist_active_record_query ||= default_active_record_query
+      end
+
+      def blacklist_active_record_query=(query_hash)
+        return unless valid_query_hash?(query_hash)
+
+        @blacklist_active_record_query = query_hash
+      end
```

Comment:
The idea is that right now this just supports `blacklists`, but in the future it could be easily extended to things like `whitelist` domains and `disposable` domains via defining the four methods above for those items as well.

Technically I could make it even easier to support new attributes via a dynamic `define_method` along with a constant of attributes I want to support dynamically, but I like explicitly outlining methods for easier debugging.